sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/FileAsyncApiTests.java

```diff
@@ -424,12 +433,3698 @@ public void createOptionsWithTimeToExpireRelativeToNow() {
+            new ParallelTransferOptions().setProgressListener(mockListener),
+            new DownloadRetryOptions().setMaxRetryRequests(3), null, false, null).block();
+
+        // Should receive at least one notification indicating completed progress, multiple notifications may be
+        // received if there are empty buffers in the stream.
+        assertTrue(mockListener.progresses.stream().anyMatch(progress -> progress == fileSize));
+
+        // There should be NO notification with a larger than expected size.
+        assertFalse(mockListener.progresses.stream().anyMatch(progress -> progress > fileSize));
+
+        // We should receive at least one notification reporting an intermediary value per block, but possibly more
+        // notifications will be received depending on the implementation. We specify numBlocks - 1 because the last
+        // block will be the total size as above. Finally, we assert that the number reported monotonically increases.
+        long prevCount = -1;
+        for (long progress : mockListener.progresses) {
+            assertTrue(progress >= prevCount, "Reported progress should monotonically increase");
+            prevCount = progress;
+        }
+    }
+
+    private static final class MockProgressListener implements ProgressListener {
+        List<Long> progresses = new ArrayList<>();
+
+        @Override
+        public void handleProgress(long progress) {
+            progresses.add(progress);
+        }
+    }
+
+    @Test
+    public void renameMin() {
+        assertAsyncResponseStatusCode(fc.renameWithResponse(null, generatePathName(),
+            null, null, null), 201);
+    }
+
+    @Test
+    public void renameWithResponse() {
+        StepVerifier.create(fc.renameWithResponse(null, generatePathName(),
+            null, null, null))
+            .assertNext(r -> {
+                StepVerifier.create(r.getValue().getPropertiesWithResponse(null))
+                    .assertNext(p -> assertEquals(p.getStatusCode(), 200))
+                    .verifyComplete();
+            })
+            .verifyComplete();
+
+        StepVerifier.create(fc.getProperties())
+            .verifyErrorSatisfies(r -> {
+                assertInstanceOf(DataLakeStorageException.class, r);
+            });
+    }
+
+    @Test
+    public void renameFilesystemWithResponse() {
+        DataLakeFileSystemAsyncClient newFileSystem = primaryDataLakeServiceAsyncClient.createFileSystem(generateFileSystemName()).block();
+
+        StepVerifier.create(fc.renameWithResponse(newFileSystem.getFileSystemName(), generatePathName(),
+            null, null, null))
+            .assertNext(r -> {
+                StepVerifier.create(r.getValue().getPropertiesWithResponse(null))
+                    .assertNext(p -> assertEquals(p.getStatusCode(), 200))
+                    .verifyComplete();
+                //assertDoesNotThrow(r.getValue()::getProperties);
+            })
+            .verifyComplete();
```

Comment:
Having a `StepVerifer` within a `StepVerifier` isn't always safe. I'd recommend a small change to this test:

```java
StepVerifier.create(fc.renameWithResponse(newFileSystem.getFileSystemName(), generatePathName(),
    null, null, null)
    .flatMap(r -> r.getValue().getPropertiesWithResponse(null)))
    .assertNext(p -> assertEquals(p.getStatusCode(), 200))
    .verifyComplete();
```

This will perform the same logic in renaming the file system and then trying to get properties using the new file system returned.